board: add support for the nucleo_f303k8 board

Gimpel-K67 opened the pull request:
This branch adds support for the nucleo_f303k8 board, the soc stmf303x8 defconfig and the stm32f303X8 dtsi file.
The configuration is based on the nucleo_l011k4, the f302r8 and the ST reverence manual.

I had successfully tested the following sample code:
blinky
blink_led (uses TIM2_CH1 on PA0)
button
hello_world

Signed-off-by: Sebastian Schwabe <sebastian.schwabe@mailbox.tu-dresden.de>

erwango commented on boards/arm/nucleo_f303k8/Kconfig.board:
```diff
@@ -0,0 +1,8 @@
+# NUCLEO-32 F303k8 board configuration
+
+# Copyright (c) 2020 Sebastian Schwabe, Fabian Paschke <sebastian.schwabe@mailbox.tu-dresden.de, fabian.paschke@eas.iis.fraunhofer.de>
```
Nit: Long line

erwango commented on boards/arm/nucleo_f303k8/Kconfig.defconfig:
```diff
@@ -0,0 +1,11 @@
+# NUCLEO-32 F303k8 board configuration
+
+# Copyright (c) 2020 Sebastian Schwabe, Fabian Paschke <sebastian.schwabe@mailbox.tu-dresden.de, fabian.paschke@eas.iis.fraunhofer.de>
```
Ditto

erwango commented on boards/arm/nucleo_f303k8/nucleo_f303k8.yaml:
```diff
@@ -0,0 +1,17 @@
+  - gpio
+  - pwm
+  - counter
+  - adc
```
i2c, spi, adc not configured in dts

erwango commented on dts/arm/st/f3/stm32f303X8.dtsi:
```diff
@@ -0,0 +1,27 @@
+	};
+
+	sram0: memory@20000000 {
+		reg = <0x20000000 DT_SIZE_K(40)>;
```
```suggestion
		reg = <0x20000000 DT_SIZE_K(12)>;
```

erwango reviewed (changes requested):
Thanks for this addition!
I only have few comments.
Also, please split your PR into boards and dts/soc commits.

Commit:
board: arm: add support for the nucleo_f303k8 board

This branch adds support for the nucleo_f303k8 board.
The configuration is based on the nucleo_l011k4,
the f302r8 and the ST reference manual.

I had successfully tested the following sample code:
blinky
blink_led (uses TIM2_CH1 on PA0)
button
hello_world

Signed-off-by: Sebastian Schwabe <sebastian.schwabe@mailbox.tu-dresden.de>

Commit:
soc: arm: st_stm32: stm32f3: Add support for STM32F303x8

This commit adds the soc config for the STM32F303x8.
Add the STM32F303x8 as choice to the Kconfig.soc.
Fixing indention error in Kconfig.soc.

Signed-off-by: Sebastian Schwabe <sebastian.schwabe@mailbox.tu-dresden.de>

TeckKitty commented:
Thanks for the additions. I split it up to three commits for soc, dts and board. It took me some time to fix the format errors but now everything look fine. I also fixed the other things you pointed out.

erwango commented on boards/arm/nucleo_f303k8/Kconfig.board:
```diff
@@ -0,0 +1,11 @@
+# NUCLEO-32 F303k8 board configuration
+
+# Copyright (c) 2020 Sebastian Schwabe
+# <sebastian.schwabe@mailbox.tu-dresden.de>
+# Fabian Paschke
+# <fabian.paschke@eas.iis.fraunhofer.de>
```
Something like follows would be nicer, but ok.
```
# Copyright (c) 2020 Sebastian Schwabe <sebastian.schwabe@mailbox.tu-dresden.de>
                                  Fabian Paschke <fabian.paschke@eas.iis.fraunhofer.de>
```

erwango commented on soc/arm/st_stm32/stm32f3/Kconfig.defconfig.stm32f303x8:
```diff
@@ -0,0 +1,17 @@
+#  Copyright (c) 2020 Sebastian Schwabe 
+#							<sebastian.schwabe@mailbox.tu-dresden.de>,
+# 					  Fabian Paschke
+#					   		<fabian.paschke@eas.iis.fraunhofer.de>
```
```suggestion
# Copyright (c) 2020 Sebastian Schwabe <sebastian.schwabe@mailbox.tu-dresden.de>
#                    Fabian Paschke <fabian.paschke@eas.iis.fraunhofer.de>
```

erwango commented on soc/arm/st_stm32/stm32f3/Kconfig.defconfig.stm32f303x8:
```diff
@@ -1,9 +1,9 @@
+#  Copyright (c) 2020 Sebastian Schwabe
+#  <sebastian.schwabe@mailbox.tu-dresden.de>
+#  Fabian Paschke
+#  <fabian.paschke@eas.iis.fraunhofer.de>
```
Please fix this in the initial commit

TeckKitty replied:
How can I change an outdated commit? I removed the initial commit and add the three new one for dts, soc and board.

erwango replied:
https://fle.github.io/git-tip-keep-your-branch-clean-with-fixup-and-autosquash.html

TL;DR:
-add a new commit ths fixes the issue. As commit title, use: `fixup! <sha of the commit to be fixed>`
-Repeat for each fix to be done on the branch
-Once you're done: `git rebase -i --autosquash HEAD~<numbers of commits of your branch>`

erwango commented:
@TeckKitty Building ADC test (tests/drivers/adc/adc_api/) requires a modification to the test.
This is not done and this is what is making CI (buildkite: https://buildkite.com/zephyr/zephyr/builds/15942) complaining.
Either you need to update the test, either you need to remove adc from the supported list in the yaml file (but you don't need to remove adc in the dts file).

Commit:
dts: arm: Add STM32F303x8 dtsi config

This commit adds the dtsi config for the stm32f303k8.
In addition fixing to long line for author.
Fixing sram0 memory size.

Signed-off-by: Sebastian Schwabe <sebastian.schwabe@mailbox.tu-dresden.de>

TeckKitty commented:
Ok I try do to the fixes you suggested to me. I mixed up the dts and soc files in one commit. Now it should be correctly separated. 
I removed the ADC from the yaml file because I have no clue how to change the test. 
All checks now passed.
It is my first time doing a merge request so thanks for your patience.
